projects/pokemon_game.py

```diff
@@ -75,65 +71,87 @@ def feed(self):
 
+charmander = Pokemon(
+    name = "Charmander", 
+    primary_type = "fire", 
+    max_hp = 50, 
+    hp = 50)
 
+snivy = Pokemon(
+    name = "Snivy", 
+    primary_type = "grass",
+    max_hp = 60, 
+    hp = 60)
 
-charmander = Pokemon("Charmander", "fire", 50, 50)
-snivy = Pokemon("Snivy", "grass", 60, 60)
-sobble = Pokemon("Sobble", "water", 60, 60)
-
-possible_pokemon = [charmander, snivy, sobble]
+sobble = Pokemon(
+    name = "Sobble", 
+    primary_type = "water",
+    max_hp = 60,
+    hp = 60)
 
-# Take user input- which Pokemon do they want to battle with?
+all_pokemon = [charmander, snivy, sobble]
 
+# User chooses a Pokemon to battle with
 def get_user_choice():
     while True:
-        user_choice = input("Which Pokemon would you like to battle with: Charmander, Snivy, or Sobble \n")
-       
-        if user_choice == charmander.name:
-            return charmander
-            # break
-        elif user_choice == snivy.name:
-            return snivy
-            # break
-        elif user_choice == sobble.name:
-            return sobble
-        else:
-            print("Please try again.")
+        print(f"Which Pokemon would you like to battle with?") 
+        for pokemon in all_pokemon:
+            print(f"{pokemon.name}")
+        user_choice = input() 
 
+        for pokemon in all_pokemon:
+            if user_choice == pokemon.name:
+                return pokemon
+
+# Get user choice of Pokemon to battle with
 user_pokemon = get_user_choice()
 print(f"You chose {user_pokemon.name} and your HP is {user_pokemon.hp}.")
 
-# Computer chooses a Pokemon to battle which is not the same as the users
-
+# Makes a list of non-zero HP and non-user-selected Pokemon
+def get_available_pokemon():
+    usable_pokemon = []
+    for pokemon in all_pokemon:
+        if pokemon.hp > 0 and pokemon.name != user_pokemon.name:
+            usable_pokemon.append(pokemon)
+      
+    print(f"Here are the other Pokemon that are alive:")
+    for item in usable_pokemon:
+        print(item.name)
+    
+    return usable_pokemon
 
+# Computer chooses a Pokemon to battle which is not the same as the users
 def get_computer_choice():
-    while True:
+        # Pick a random Pokemon from the list
+        possible_pokemon = get_available_pokemon()
         computer_choice = random.choice(possible_pokemon)
-        # print(f"comp: {computer_choice.name}")
-        if computer_choice.name != user_pokemon.name:
-            print(f"The computer has chosen to battle with {computer_choice.name} and their HP is {computer_choice.hp}.")
-            return computer_choice
+        print(f"The computer has chosen to battle with {computer_choice.name} and their HP is {computer_choice.hp}.")
+        return computer_choice
 
+# Get computer pokemon
 computer_pokemon = get_computer_choice()
 
-
-# Call Battle
+# Call Battle the battle between the user and the computer
 user_pokemon.battle(computer_pokemon)
 
 
 while True:
```

Comment:
**suggestion**: Let's create a `game_loop()` function that can contain pretty much all of the code in the `while True` loop, just to keep things nice an organized.